MediaGalleryUi/Model/FilterParametersProvider/Fulltext.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\MediaGalleryUi\Model\FilterParametersProvider;
+
+use Magento\Framework\Api\SearchCriteriaInterface;
+use Magento\Framework\DB\Adapter\AdapterInterface;
+use Magento\Framework\DB\Select;
+use Magento\MediaGalleryUi\Model\SelectModifierInterface;
+
+/**
+ * Apply fulltext filter
+ */
+class Fulltext implements SelectModifierInterface
```

Comment:
@sivaschenko  Maybe it will be better to granulate this class into 2: `Title` and `Keywords`. It is too abstract to name it `Fulltext` and collect only title and keywords. As I can see it is also good to add full text on `adobe_stock_category` and `adobe_stock_creator` if we use adobe stock asset info.

Reply:
The idea is to add fulltext conditions using `OR` condition, so a single class should be responsible for handling the fulltext filter. We could introduce the extension point for other fields applicable for fulltext filter later. Let's keep this implementation as it is right now